defaults/validate.go

```diff
@@ -123,6 +123,14 @@ func validate(validation, fieldName string, p, v reflect.Value) error {
 		if s == "" || !isAlphanumeric(s) {
 			return fmt.Errorf("%s is not alphanumeric", fieldName)
 		}
+	case "emptyorfile":
```

Comment:
Given the license is the only use of the `file` tag I wonder if we could more simply make it accept empty values. It would be cool to allow multiple subtags, so one could define something like `validate:"notempty,file"`.